File: jinxExcel/translateRule/mission.py
import sys
import os
sys.path.append(os.path.dirname(__file__)+'/'+'..')

import parseHelper.LuaType as lt
import parseHelper.RuleType as rt


RULES=rt.NormalRule({
    "任务发布点" :{"to" : "belongMissionPlace" , "type" : lt.LuaStr },
    "任务名" :{"to" : "missionName" , "type" : lt.LuaStr },
    "描述" :{"to" : "desc" , "type" : lt.LuaBlockStr },
    "任务完成描述" :{"to" : "completeDesc" , "type" : lt.LuaBlockStr },
    "完成获得标志" :{"to" : "tag" , "type" : lt.LuaStr },
    "前置条件" : {"to" : "condition" ,"type" : lt.LuaParseStr ,"split" : "\n:", "rule" :{
        0 : {"to" : "type", "type" : lt.LuaStr},
        1 : {"to" : "value", "type" : lt.LuaStr},
    }},
    "可接取次数" :{"to" : "limit" , "type" : lt.LuaInt },
    "extend1":rt.ExtendRule(1,5,"need","条件[index]类型","条件[index]值",{
            "条件[index]类型": {"to" : "type" ,"type" : lt.LuaStr},
            "条件[index]值": {"to" : "value" ,"type" : lt.LuaStr},
    }),
    "extend2":rt.ExtendRule(1,5,"cost","消耗[index]类型","消耗[index]值",{
            "消耗[index]类型": {"to" : "type" ,"type" : lt.LuaStr},
            "消耗[index]值": {"to" : "value" ,"type" : lt.LuaStr},
    }),
    "extend3":rt.ExtendRule(1,5,"award","奖励[index]类型","奖励[index]值",{
            "奖励[index]类型": {"to" : "type" ,"type" : lt.LuaStr},
            "奖励[index]值": {"to" : "value" ,"type" : lt.LuaStr},
    }),
})

BUILD=\
[
    #构造任务
    {
        #构造目标
        "target" : "mission",
        #源文件
        "source": ["R任务.xlsx"],
        #sheet对应的构造规则
        "sheet": [
            ("任务" , RULES)
        ]    
    },

]

File: gameDebug/src/resScripts/time1/mission.lua
local mission={

{ belongMissionPlace="大理.有间客栈.任务",missionName="找野兽皮",desc=[=[宁静村的老伯要为过冬准备一件衣服,他希望得到足够的兽皮]=],completeDesc=[=[找到足够兽皮了,快去宁静村找老伯吧]=],tag="任务.大理.找兽皮",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="item",value="兽皮,3" } },
cost={ 
{ type="item",value="兽皮,3" } } },

{ belongMissionPlace="大理.有间客栈.任务",missionName="找银手镯",desc=[=[段王府的小孩丢了一个银手镯,希望你能给他一个银手镯]=],completeDesc=[=[找到了银手镯,去段王府找找小孩]=],tag="任务.大理.找手镯",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="item",value="银手镯,1" } },
cost={ 
{ type="item",value="银手镯,1" } } },

{ belongMissionPlace="大理.有间客栈.任务",missionName="除恶虎",desc=[=[最近乱葬岗附近有恶虎伤人,希望你能除掉它]=],completeDesc=[=[已经除掉了恶虎快回客栈领取报酬吧]=],tag="任务.大理.除虎",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="kill",value="乱葬岗恶虎,1" } },
cost={  } },

{ belongMissionPlace="大理.有间客栈.任务",missionName="三次元",desc=[=[这附近有两个友情客串的npc,请找到他们.]=],completeDesc=[=[已经找到了两个客串npc,快去客栈领取奖励吧]=],tag="任务.大理.三次元",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="meet",value="待定" },
{ type="meet",value="扎西" } },
cost={  } },

{ belongMissionPlace="大理.有间客栈.任务",missionName="大战木头人",desc=[=[你太菜了,需要打败木头人]=],completeDesc=[=[已经打败了木头人,快去客栈领取奖励吧]=],tag="任务.大理.木头人",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="kill",value="大理木头人,1" } },
cost={  } },

{ belongMissionPlace="大理.有间客栈.任务",missionName="完成以上所有任务",desc=[=[完成以上所有任务,将开启世界剧情]=],completeDesc=[=[任务已经完成,快回客栈吧.]=],tag="任务.大理.完成以上所有任务",
condition={ 
{ type="CHAPTER",value="俗世凡尘" } },limit=1,
need={ 
{ type="complete",value="大理.有间客栈.任务,找野兽皮" },
{ type="complete",value="大理.有间客栈.任务,找银手镯" },
{ type="complete",value="大理.有间客栈.任务,除恶虎" },
{ type="complete",value="大理.有间客栈.任务,三次元" } },
cost={  } },

}
return mission